### PR TITLE
add transport docs in readme

### PR DESCRIPTION
**Pull request by xiangyan99**

### CODE REVIEW

File: sdk/core/azure-core/README.md [hunk 1]
```diff
@@ -112,6 +112,28 @@ class TooManyRedirectsError(HttpResponseError):
 
 *kwargs* are keyword arguments to include with the exception.
 
+### Transport
+
+Some common properties can be configured on transports. They must be passed
+as kwargs arguments while building the transport instance.
+
+#### Transport configurations
```
Comment (lmazuel) [inline sdk/core/azure-core/README.md:120]: I don't like duplication as you know, so we should then remove that:
https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md#transport-configurations

We need to decide if its interesting for both user and dev (this file), or just dev (the client_library_dev file), but not both.

Comment (xiangyan99) [reply sdk/core/azure-core/README.md:120]: IMO, the most valuable information for user is

#### Async transport

The async transport is designed to be opt-in. [AioHttp](https://pypi.org/project/aiohttp/) is one of the supported implementations of async transport. It is not installed by default. You need to install it separately.

for transport kwargs, we can put them together with policy kwargs.

How about you? @chlowell

Comment (chlowell) [reply sdk/core/azure-core/README.md:120]: Seems reasonable to me. The kwargs are part of the public API, and users are most impacted by the implicit aiohttp dependency, so it makes sense to include them in documentation written for users.

**Review by lmazuel — APPROVED** [review]

**Pull request merged**